File: Java/anagramCheck.java
import java.util.Arrays;

/**
 
Anagram Check
Problem Statement: Check if two given strings are anagrams of each other.
Input: "listen", "silent"
Expected Result: True

 */
public class anagramCheck {

    public static void main(String[] args) {
        String first="listen";
        String second= "silent";
        char[] fc=first.toCharArray();
        char[] sc=second.toCharArray();
        Arrays.sort(fc);
        Arrays.sort(sc);
        if(Arrays.equals(fc, sc)){
            System.out.println("True");


        }
        else{
            System.out.println("False");
        }


    }

    
}

File: DSAPython/anagramCheck.py
"""Anagram Check
Problem Statement: Check if two strings are anagrams.
Input: "listen", "silent"
Expected Result: True"""

str1="listen"
str2="silent"

dict1={}
dict2={}

for ch in  str1:
    dict1[ch]=dict1.get(ch,0)+1
    
for ch in str2:
    dict2[ch]= dict2.get(ch,0)+1
    
print(dict1==dict2)

